sdk/search/search-documents/src/indexModels.ts

```diff
@@ -685,3 +722,136 @@ export type IndexDocumentsAction<T> = {
 } & Partial<T>;
 
 // END manually modified generated interfaces
+
+// Utility types
+
+export type UnionToIntersection<U> =
+  // Distribute members of U into parameter position of a union of functions
+  (
+    U extends unknown ? (_: U) => unknown : never
+  ) extends // Infer the intersection of the members of U as a single intersected parameter type
+  (_: infer I) => unknown
+    ? I
+    : never;
+
+// Types that should not be included in SelectFields recursion
+export type ExcludedODataTypes = Date | GeographyPoint;
+
+/**
+ * Produces a union of valid Cognitive Search OData $select paths for T
+ * using a post-order traversal of the field tree rooted at T.
+ */
+export type SelectFields<T extends object> = T extends Array<infer U>
+  ? // Allow selecting fields only from elements which are objects
+    NonNullable<U> extends object
+    ? SelectFields<NonNullable<U>>
+    : never
+  : {
+      // Only consider string keys
+      [K in Exclude<keyof T, symbol | number>]: NonNullable<T[K]> extends object
```

Comment:
This should be equivalent and somewhat simpler:
```suggestion
      [K in keyof T & string]: NonNullable<T[K]> extends object
```